test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -272,6 +274,29 @@ TEST(QuicPlatformTest, QuicNotification) {
   EXPECT_TRUE(notification.HasBeenNotified());
 }
 
+TEST(QuicPlatformTest, QuicCertUtils) {
+  bssl::UniquePtr<X509> x509_cert =
+      TransportSockets::Tls::readCertFromFile(TestEnvironment::substitute(
+          "{{ test_rundir }}/test/extensions/transport_sockets/tls/test_data/san_dns_cert.pem"));
+  // Encode X509 cert with DER encoding.
+  int len = i2d_X509(x509_cert.get(), nullptr);
+  unsigned char* der = static_cast<unsigned char*>(OPENSSL_malloc(len));
+  unsigned char* p = der;
+  i2d_X509(x509_cert.get(), &p);
+  EXPECT_EQ(p - der, len);
```

Comment:
You can make this a hair tighter (and handle errors) by replacing lines 282 through 286 with:

```
  unsigned char* der = nullptr;
  int len = i2d_X509(x509_cert.get(), &der);
  ASSERT_GT(len, 0);
```

The `i2d_*` functions are super goofy and treat pointer to null pointer as a totally separate calling convention where they `OPENSSL_malloc` a buffer for you. (Yes, this is ridiculous. Those APIs are awful. :-/ )

Reply:
Good to know! Done